src/builtins/math.rs

```diff
@@ -132,7 +163,12 @@ fn format_double(mut v: f64, opts: &Options) -> WString {
     // As a special-case, a scale of 0 means to truncate to an integer
     // instead of rounding.
     if opts.scale == 0 {
-        v = v.trunc();
+        v = match opts.zero_scale_mode {
```

Comment:
So the way this works right now is that you can do `math --scale-mode=round`, and it will do absolutely nothing because you forgot to set `--scale=0` as well.

I would make it so setting scale-mode implies a scale of 0, and explicitly giving any *other* scale is an error (for now, it could potentially be changed later)

Reply:
An alternative is to have a special argument to `--scale`, something like `--scale=0-truncate`.

Reply:
I'm not very familiar with this project and want to know if there are something like 'log level' in fish shell? If so, I want to make use of that.
Of course the default behavior is leaving error to case above. But it will also leave room for intended behavior and only throw warning or even nothing.

Reply:
> Of course the default behavior is leaving error to case above

I'm saying that `--scale-mode=foo` should automatically set `--scale=0`, and error out if you gave a non-zero scale value because that's nonsensical.

That's specifically an error in how it was called, and it should be an error.

> only throw warning or even nothing

Warnings aren't a good idea here.

We have [a logging system](https://github.com/fish-shell/fish-shell/blob/master/src/flog.rs) for fish-internal things, but it's not something you would use for builtins - which should mostly be treated as if they were standalone programs.

Reply:
Got it